frontend/src/metabase/nav/components/search/RecentsList/RecentsListContent.tsx

```diff
@@ -80,7 +80,7 @@ export const RecentsListContent = ({
                   <ResultTitle
                     data-testid="recently-viewed-item-title"
                     truncate
-                    href={getItemUrl(item) ?? undefined}
+                    href={!onClick ? getItemUrl(item) : undefined}
```

Comment:
The logic is simpler without the negation:

```suggestion
                    href={onClick ? undefined: getItemUrl(item)}
```